zeebe/gateway-grpc/src/main/java/io/camunda/zeebe/gateway/RequestMapper.java

```diff
@@ -163,7 +166,45 @@ private static JobResult getJobResultOrDefault(final CompleteJobRequest request)
     if (!request.hasResult()) {
       return null;
     }
-    return new JobResult().setDenied(request.getResult().getDenied());
+
+    if (!request.getResult().hasCorrections()) {
+      return new JobResult().setDenied(request.getResult().getDenied());
+    }
+
+    final JobResultCorrections corrections = new JobResultCorrections();
+    final List<String> correctedAttributes = new ArrayList<>();
+
+    if (request.getResult().getCorrections().hasAssignee()) {
+      corrections.setAssignee(request.getResult().getCorrections().getAssignee());
+      correctedAttributes.add(UserTaskRecord.ASSIGNEE);
+    }
+    if (request.getResult().getCorrections().hasDueDate()) {
+      corrections.setDueDate(request.getResult().getCorrections().getDueDate());
+      correctedAttributes.add(UserTaskRecord.DUE_DATE);
+    }
+    if (request.getResult().getCorrections().hasFollowUpDate()) {
+      corrections.setFollowUpDate(request.getResult().getCorrections().getFollowUpDate());
+      correctedAttributes.add(UserTaskRecord.FOLLOW_UP_DATE);
+    }
+    if (request.getResult().getCorrections().hasCandidateUsersList()) {
+      corrections.setCandidateUsersList(
+          request.getResult().getCorrections().getCandidateUsersList().getValuesList());
+      correctedAttributes.add(UserTaskRecord.CANDIDATE_USERS);
+    }
+    if (request.getResult().getCorrections().hasCandidateGroupsList()) {
+      corrections.setCandidateGroupsList(
+          request.getResult().getCorrections().getCandidateGroupsList().getValuesList());
+      correctedAttributes.add(UserTaskRecord.CANDIDATE_GROUPS);
+    }
+    if (request.getResult().getCorrections().hasPriority()) {
+      corrections.setPriority(request.getResult().getCorrections().getPriority());
+      correctedAttributes.add(UserTaskRecord.PRIORITY);
+    }
```

Comment:
🔧 What do you think about extracting `request.getResult().getCorrections()` into a variable to improve readability and reduce repetition?

For example:

```java
final GatewayOuterClass.JobResultCorrections requestCorrections = request.getResult().getCorrections();

// or -> final var requestCorrections = request.getResult().getCorrections();

if (requestCorrections.hasAssignee()) {
  corrections.setAssignee(requestCorrections.getAssignee());
  correctedAttributes.add(UserTaskRecord.ASSIGNEE);
}
if (requestCorrections.hasDueDate()) {
  corrections.setDueDate(requestCorrections.getDueDate());
  correctedAttributes.add(UserTaskRecord.DUE_DATE);
}
// and so on...
```

This might help reduce redundancy and make the code's intent clearer.
Curious to hear your thoughts! 😊

Reply:
Yes, good proposal 👍 
Thank you, @ce-dmelnych

Reply:
I didn't use `final var requestCorrections ` to be consistent with previous declarations.